Tag/WebScrap.Tags/Tools/AttributeExtractor.cs

```diff
@@ -2,34 +2,81 @@ namespace WebScrap.Tags.Tools;
 
 internal static class AttributeExtractor
 {
-    internal static ILookup<string, string> EmptyAttributes 
-        => Array.Empty<int>().ToLookup(x => "", x => "");
+    internal static ILookup<string, string?> EmptyAttributes 
+        => Array.Empty<int>().ToLookup(x => "", x => (string?)null);
 
-    internal static ILookup<string, string> GetAttributes(
+    internal static ILookup<string, string?> GetAttributes(
         ReadOnlySpan<char> tagContent)
     {
         tagContent = tagContent.TrimEnd('/');
-        var key = GetKey(tagContent).ToString();
-        var values =  GetValues(tagContent); 
 
-        return values?.ToLookup(x => key, x => x)
-            ?? EmptyAttributes;   
+        var keyValues = new List<KeyValuePair<string, string?>>();
+        while (tagContent.IsEmpty is false)
+        {         
+            tagContent = tagContent.TrimStart(' ');   
+            var key = GetKey(tagContent).ToString();
+            var values = GetValues(tagContent);
+
+            if (values is null)
+            {
+                keyValues.Add(new KeyValuePair<string, string?>(key, null));
+            }
+            else
+            {
+                var lookup = values.Select(v => new KeyValuePair<string, string?>(key, v))
+                .ToArray();
+                keyValues.AddRange(lookup);
+            }
+                
+            var delimeterOffset = values switch 
+            {
+                null => 0,
+                _ => 2,
+            };
+
+            var offset = tagContent
+                [key.Length..]
+                [delimeterOffset..];
+
+            tagContent = values switch 
```

Comment:
too much of values == null checks.
Please extract in two separate ways, for values and values-less parsing.